### PR TITLE
CXX-920 Add error reporting methods to env

### PR DESCRIPTION
**Pull request by jrassi**
Server SHAs cherry-picked (with modifications) into this commit:

1bb3be7

### CODE REVIEW

**Comment by jrassi** [comment]
Only a few hunks applied automatically, so the majority of this merge was manual.

@acmorrow, PTAL.

File: SConstruct [hunk 1]
```diff
@@ -599,11 +608,9 @@ else:
     env['OS_FAMILY'] = 'posix'
 
 if has_option( "cc-use-shell-environment" ) and has_option( "cc" ):
-    print("Cannot specify both --cc-use-shell-environment and --cc")
-    Exit(1)
+    env.FatalError("Cannot specify both --cc-use-shell-environment and --cc")
 elif has_option( "cxx-use-shell-environment" ) and has_option( "cxx" ):
-    print("Cannot specify both --cxx-use-shell-environment and --cxx")
-    Exit(1)
+    env.FatalError("Cannot specify both --cxx-use-shell-environment and --cxx")
```
Comment (acmorrow) [inline SConstruct]: Can't comment on the right line, but please see 628 and 633 below

**Comment by acmorrow** [comment]
I think I see two places that didn't get updated.

Also, did you check the various SConscript files as well to see if any of them have any Exit/exit branches that could be patched up?

**Comment by acmorrow** [comment]
LGTM mod checking other SConscripts

**Comment by jrassi** [comment]
Those two calls fixed in bd73709.  The lowercasing of 'exit' there threw off my grep invocation.

Yep, checked other SConscript files. `find . -name 'SC*' | xargs grep -n -i 'exit('` returns only lines from SConstruct above line 530 (around where the declaration of `env` is).

**Comment by acmorrow** [comment]
@jrassi - Thanks, definitely LGTM. Should definitely make future backports safer.

**Pull request closed without merging**